services/dedup/scylla/scylla.go

```diff
@@ -0,0 +1,93 @@
+package scylla
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/services/dedup/types"
+)
+
+type ScyllaDB struct {
+	scylla         *gocql.Session
+	conf           *config.Config
+	stat           stats.Stats
+	ttl            int
+	batchSize      int
+	createTableMap map[string]struct{}
+}
+
+func (d *ScyllaDB) Close() {
+	d.scylla.Close()
+}
+
+func (d *ScyllaDB) Set(kv types.KeyValue) (bool, int64, error) {
+	// Create the table if it doesn't exist
+	if _, ok := d.createTableMap[kv.WorkspaceId]; !ok {
```

Comment:
We can consider using [sync once values](https://pkg.go.dev/sync#OnceValues) here and possibly some locking for the map. The code may run multiple times if multiple workspaceIds access the same time.

Reply:
Not sure how we could use sync once without using a map. Do you have a sample code in mind?

Reply:
Something like this should work, right? https://go.dev/play/p/kgVwQclcxIy